python/nano/src/bigdl/nano/pytorch/onnx/onnxrt_inference.py

```diff
@@ -0,0 +1,199 @@
+                       input_sample=None,
+                       file_path="model.onnx",
+                       sess_options=None,
+                       **kwargs):
+        '''
+        Internal function to build a ortsess and bind to the lightningmodule.
+
+        :param input_sample: torch.Tensor for the model tracing.
+        :param file_path: The path to save onnx model file.
+        :param sess_options: ortsess options in ort.SessionOptions type
+        :param **kwargs: will be passed to torch.onnx.export function.
+        '''
+
+        if input_sample is None and self.example_input_array is not None:
+            input_sample = self.example_input_array  # use internal example_input_array
+        else:
+            self.example_input_array = input_sample  # set example_input_array for future usage
+
+        assert input_sample is not None,\
+            'You should set either input_sample or self.example_input_array'
+
+        default_onnx_export_args = {'export_params': True,
+                                    'opset_version': 10,
+                                    'do_constant_folding': True,
+                                    'input_names': ['input'],
+                                    'output_names': ['output'],
+                                    'dynamic_axes': {'input': {0: 'batch_size'},
+                                                     'output': {0: 'batch_size'}}}
+        default_onnx_export_args.update(kwargs)
+
+        self.to_onnx(file_path,
+                     input_sample,
+                     **default_onnx_export_args)
+
+        self._ortsess = ort.InferenceSession(file_path, sess_options=sess_options)
+        self._ortsess_up_to_date = True
+
+    pl_model._build_ortsess = partial(_build_ortsess, pl_model)
+
+    # external method to update(& rebuild) ortsess
+    def update_ortsess(self,
+                       input_sample=None,
+                       file_path="model.onnx",
+                       sess_options=None,
+                       **kwargs):
+        '''
+        Update the onnxruntime session options and rebuild the session.
+        Users may also want to call this method before `inference(..., onnx=True`)`
+        to avoid implicit building.
+
+        :param input_sample: torch.Tensor for the model tracing.
+        :param file_path: The path to save onnx model file.
+        :param sess_options: ortsess options in ort.SessionOptions type.
+        :param **kwargs: will be passed to torch.onnx.export function.
+        '''
+        self._build_ortsess(input_sample=input_sample,
+                            file_path=file_path,
+                            sess_options=sess_options,
+                            **kwargs)
+
+    pl_model.update_ortsess = partial(update_ortsess, pl_model)
+
+    # on_fit_start (LightningModule method override)
+    def on_fit_start_additional(function):
+        def wrapped(*args, **kwargs):
+            args[0]._ortsess_up_to_date = False
```

Comment:
shall we also clean up and release the resources here?

Reply:
good idea, I will release the resource here